zim/gui/pageview.py

```diff
@@ -4863,22 +4897,61 @@ def discard(self):
 			pageview.set_page(page)
 			self._done = True
 
+		def discard_disk(self):  
+                        logger.debug('allowing to overwrite the file')
+			text, page._last_etag = page.source_file.read_with_etag()
+                        page._store_last_lines(page.format.Dumper().dump(self.get_parsetree(),file_output=True))
```

Comment:
Found the culprit for the "discard disk" button not working: change this line to:

    page._store_last_lines(page.format.Dumper().dump(page.get_parsetree(), file_output=True))

Reply:
Also could you add a few comments to explain what each line does? The first line seems to be the one that keeps the buffer and discards the disk, so why do we need the other lines of code?